src/components/container/AIAssistant/AIAssistant.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import { faUser } from '@fortawesome/free-solid-svg-icons/faUser';
+import { faGear } from '@fortawesome/free-solid-svg-icons/faGear';
+import { StreamEvent } from '@langchain/core/tracers/log_stream';
+import { AIMessageChunk } from '@langchain/core/messages';
+import { StructuredTool } from '@langchain/core/tools';
+
+import { MyDataHelpsAssistant } from '../../../helpers/assistant/assistant';
+import Chat from '../../presentational/Chat';
+
+import '@fortawesome/fontawesome-svg-core/styles.css';
+
+export interface AIAssistantProps {
+    innerRef?: React.Ref<HTMLDivElement>;
+    previewState?: "default";
+    debug: boolean;
+    additionalInstructions?: string;
+    tools?: StructuredTool[];
+    appendTools?: boolean;
+}
+
+export type AIAssistantMessageType = "user" | "ai";
+
+export interface AIAssistantMessage {
+    type: AIAssistantMessageType;
+    content: string;
+    runId?: string;
+}
+
+export default function (props: AIAssistantProps) {
+
+    const [messages, setMessages] = useState<AIAssistantMessage[]>([]);
+    const [loading, setLoading] = useState("");
+
+    const assistantRef = useRef<MyDataHelpsAssistant>();
+
+    useEffect(() => {
+        if (assistantRef.current === undefined) {
```

Comment:
Out of curiosity why ref and not state?

Reply:
Conceptually seems slightly different. https://react.dev/reference/react/useRef this is close to the `interval` and `VideoPlayer` examples in the documentation.